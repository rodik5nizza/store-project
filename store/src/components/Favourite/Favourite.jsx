import styles from './Favourite.module.scss';
import PropTypes from "prop-types";

const Favourite = ({counter}) => {
    return(
        <div className={styles.favouriteWrapper}>
            <svg xmlns="http://www.w3.org/2000/svg"  height="40px" width="40px" viewBox="0 0 24 24"> <g> <path fill="none" d="M0 0h24v24H0z"/> <path fill="yellow" d="M12 17l-5.878 3.59 1.598-6.7-5.23-4.48 6.865-.55L12 2.5l2.645 6.36 6.866.55-5.231 4.48 1.598 6.7z"/> </g> </svg>
            {counter > 0 && <div className={styles.counter}>{counter}</div>}
        </div>
    )
}

Favourite.propTypes = {
	counter: PropTypes.number,
}

Favourite.defaultProps = {
	counter: 0,
}

export default Favourite