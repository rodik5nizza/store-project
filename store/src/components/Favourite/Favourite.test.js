import Favourite from "./Favourite";
import { render } from "@testing-library/react";


describe('test snapshot Favourite', () => {
    test('should Favourite match snapshot', () => {
        const { asFragment } = render(<Favourite counter={1}/>)
        expect(asFragment()).toMatchSnapshot()
    })
})