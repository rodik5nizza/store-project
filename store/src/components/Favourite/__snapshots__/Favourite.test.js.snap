// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot Favourite should Favourite match snapshot 1`] = `
<DocumentFragment>
  <div
    class="favouriteWrapper"
  >
    <svg
      height="40px"
      viewBox="0 0 24 24"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
       
      <g>
         
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
         
        <path
          d="M12 17l-5.878 3.59 1.598-6.7-5.23-4.48 6.865-.55L12 2.5l2.645 6.36 6.866.55-5.231 4.48 1.598 6.7z"
          fill="yellow"
        />
         
      </g>
       
    </svg>
    <div
      class="counter"
    >
      1
    </div>
  </div>
</DocumentFragment>
`;
